air/src/stack/op_flags/tests.rs

```diff
@@ -121,3 +121,410 @@ fn degree_4_op_flags() {
         }
     }
 }
+
+/// Composite flag unit tests.
+#[test]
+fn composite_flags() {
+    // ------ no change 0 ---------------------------------------------------------------------
+
+    let op_no_change_0 = [Operation::MpVerify, Operation::Span, Operation::Halt];
+    for op in op_no_change_0 {
+        // frame initialised with an op operation.
+        let frame = generate_evaluation_frame(op.op_code().into());
+
+        // All the operation flags are generated for the given frame.
+        let op_flags = OpFlags::new(&frame);
+
+        for i in op_flags.no_shift_flags.iter() {
+            assert_eq!(*i, ONE);
+        }
+
+        for i in op_flags.left_shift_flags.iter() {
+            assert_eq!(*i, ZERO);
+        }
+
+        for i in op_flags.right_shift_flags.iter() {
+            assert_eq!(*i, ZERO);
+        }
+
+        assert_eq!(op_flags.right_shift_flag, ZERO);
+        assert_eq!(op_flags.left_shift_flag, ZERO);
+
+        if op == Operation::MpVerify {
+            assert_eq!(op_flags.control_flow_flag, ZERO);
+        } else if op == Operation::Span || op == Operation::Halt {
+            assert_eq!(op_flags.control_flow_flag, ONE);
+        } else {
+        }
+    }
+
+    // ------ No change 1 ---------------------------------------------------------------------
+
+    let op = Operation::Incr;
+    // frame initialised with an op operation.
+    let frame = generate_evaluation_frame(op.op_code().into());
+
+    // All the operation flags are generated for the given frame.
+    let op_flags = OpFlags::new(&frame);
+
+    for (i, ele) in op_flags.no_shift_flags.iter().enumerate() {
+        if i == 0 {
+            assert_eq!(*ele, ZERO);
+        } else {
+            assert_eq!(*ele, ONE);
+        }
+    }
```

Comment:
nit: a slightly more readable way would be to check the first flag outside of the loop, and the all other ones inside a loop.